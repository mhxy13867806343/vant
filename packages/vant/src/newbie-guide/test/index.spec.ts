import { later } from '../../../test';
import { mount } from '../../../test';
import { NewbieGuide } from '..';

test('should render newbie-guide correctly', async () => {
  const wrapper = mount(NewbieGuide, {
    props: {
      show: true,
      steps: [
        {
          target: '.test-target',
          content: 'Test Content',
        },
      ],
    },
  });

  await later();
  expect(wrapper.html()).toMatchSnapshot();
});

test('should emit close event when clicking overlay', async () => {
  const wrapper = mount(NewbieGuide, {
    props: {
      show: true,
      closeOnClickOverlay: true,
      steps: [
        {
          target: '.test-target',
          content: 'Test Content',
        },
      ],
    },
  });

  await later();
  wrapper.find('.van-newbie-guide__mask').trigger('click');
  expect(wrapper.emitted('update:show')?.[0]).toEqual([false]);
  expect(wrapper.emitted('close')?.[0]).toBeTruthy();
});

test('should emit step-change event when clicking next button', async () => {
  const wrapper = mount(NewbieGuide, {
    props: {
      show: true,
      steps: [
        {
          target: '.test-target-1',
          content: 'Step 1',
        },
        {
          target: '.test-target-2',
          content: 'Step 2',
        },
      ],
    },
  });

  await later();
  wrapper.find('.van-newbie-guide__next').trigger('click');
  expect(wrapper.emitted('step-change')?.[0]).toEqual([1]);
});
