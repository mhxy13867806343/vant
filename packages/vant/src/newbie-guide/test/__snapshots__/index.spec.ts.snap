// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`should render newbie-guide correctly 1`] = `
<div class="van-newbie-guide">
  <div class="van-newbie-guide__mask">
  </div>
  <div
    class="van-newbie-guide__content"
    style="z-index: 100;"
  >
    <div class="van-newbie-guide__step-content">
      Test Content
    </div>
    <div class="van-newbie-guide__buttons">
      <button class="van-newbie-guide__next van-theme-primary">
        完成
      </button>
    </div>
  </div>
</div>
`;
